pkg/ingester/flush_test.go

```diff
@@ -103,12 +149,19 @@ func (s *testStore) Put(ctx context.Context, chunks []chunk.Chunk) error {
 	if err != nil {
 		return err
 	}
-	for _, chunk := range chunks {
+	for ix, chunk := range chunks {
 		for _, label := range chunk.Metric {
 			if label.Value == "" {
 				return fmt.Errorf("Chunk has blank label %q", label.Name)
 			}
 		}
+
+		// remove __name__ label
+		if chunk.Metric.Has("__name__") {
```

Comment:
you could use labels.MetricName (https://github.com/prometheus/prometheus/blob/master/pkg/labels/labels.go#L30)